docs/UserGuide.adoc

```diff
@@ -377,57 +412,83 @@ TODO: ADD DETAILS ONCE IMPLEMENTED
 [[FilterStudent]]
 ==== Filter `filter`
 
-TODO: ADD DETAILS ONCE IMPLEMENTED
+Filter Student returns you all the students according to the given keywords below.
+
+You can filter students in a class using these two ways:
+
+****
+1. module code and group code `m/` `g/`
+2. module code `m/`
+****
+
+1. Using both module code and group code returns you all the students specified by the group code and module code.
```

Comment:
Using both module code and group code will show you the students inside the group with the given group code of the module specified by the given module code.